### PR TITLE
fixes #450

### LINKED ISSUE #450: Players should not be able to set themselves unavailable when their game is already scheduled
Affects only the team league as in the individual competitions players cannot change the availability of the ongoing round anyway. Related to #236 but independent. #236 handles the situation where a player is set unavailable and then the game is scheduled, this issue handles the situation when the game is scheduled and then the player tries to go unavailable.

### PR DESCRIPTION
**Pull request by hopffgam**
fixes #450 
if players already have a scheduled game for a round they cannot set
themselves unavailable anymore.
This only affects the team league directly, as in lonewolf and chess960
you cannot set yourself unavailable for an ongoing round anyway.

### CODE REVIEW

File: heltour/tournament/api.py [hunk 1]
```diff
@@ -375,6 +375,7 @@ def set_availability(request):
         round_ = _get_next_round(league_tag, season_tag, round_num)
         player = Player.objects.filter(lichess_username__iexact=player_name).first()
         season_player = SeasonPlayer.objects.filter(player=player, season=round_.season).first()
+
```
Comment (lakinwecker) [inline heltour/tournament/api.py:378]: Why the extra line?

Comment (hopffgam) [reply heltour/tournament/api.py]: fixed

File: heltour/tournament/api.py [hunk 2]
```diff
@@ -386,6 +386,12 @@ def set_availability(request):
     
     if season_player.card_color == "red":
         return JsonResponse({'updated': 0, 'error': 'player_red_card'})
+    
+    if season_player.has_scheduled_game_in_round(round_):
+        return JsonResponse({'updated': 0, 'error': 'scheduled_game'})
+    
+    if has_scheduled_game:
```
Comment (lakinwecker) [inline heltour/tournament/api.py:393]: Where is this variable set? Why is the error for this clause different from the error in the one above?

File: heltour/tournament/templates/tournament/availability.html [hunk 3]
```diff
@@ -45,13 +45,13 @@ <h3>Edit Availability</h3>
                                 <tr>
                                     <td>Round {{ r.number }}</td>
                                     <td>{{ r.start_date|date_or_q:'%b %-d' }} - {{ r.end_date|date_or_q:'%b %-d' }}</td>
-                                    {% for p, is_av, is_red in av_list %}
+                                    {% for p, is_av, is_fixed in av_list %}
```
Comment (lakinwecker) [inline heltour/tournament/templates/tournament/availability.html:48]: I don't remember what this variable was supposed to indicate in the previous version, but the new name also doesn't seem particularly clear to me. 

Is it necessary for this branch?

File: heltour/tournament/views.py [hunk 4]
```diff
@@ -1839,14 +1839,20 @@ def view(self, post=False):
             season_player_set = SeasonPlayer.objects.filter(player__in=player_list, season=self.season).nocache()
             has_red_card_dict = {sp.player : sp.card_color == "red" for sp in season_player_set}
             
+            if self.league.competitor_type == 'team':
```
Comment (lakinwecker) [inline heltour/tournament/views.py:1842]: We use this kind of if statement twice in two different places, probably a candidate for refactoring. 

```python
def is_team_league(self):
    ...
``` or something.

**Comment by lakinwecker** [comment]
Pointed out one thing that I believe is a bug, and a few other minor cleanups that I think are worth doing. Ping me when they're done and I'll review it again.

File: heltour/tournament/templates/tournament/availability.html (hunk 3 above)
Comment (hopffgam) [reply heltour/tournament/templates/tournament/availability.html:48]: That parameter controls if the availability checkbox can be changed or not. I have renamed it to "cannot_change" in the upcoming commit. I know that using negations in boolean variables is usually not very elegant but here I think it conveys the usage.

File: heltour/tournament/views.py (hunk 4 above)
Comment (hopffgam) [reply heltour/tournament/views.py:1842]: I defined that function and and replaced those competitor_type=='team' calls, but I limited myself to those files that I had to touch anyway (views.py and models.py)

File: heltour/tournament/templates/tournament/availability.html (hunk 3 above)
Comment (lakinwecker) [reply heltour/tournament/templates/tournament/availability.html]: What about `is_readonly` or `is_immutable`.  I don't mind `cannot_change`, but these are other potential names. I'm happy to merge it as is, though.

Comment (hopffgam) [reply heltour/tournament/templates/tournament/availability.html]: I like is_immutable bebest and will adapt it.

**Pull request merged**